Mathlib/GroupTheory/OrderOfElement.lean

```diff
@@ -118,6 +118,37 @@ theorem isOfFinOrder_one : IsOfFinOrder (1 : G) :=
 #align is_of_fin_order_one isOfFinOrder_one
 #align is_of_fin_order_zero isOfFinAddOrder_zero
 
+open Submonoid in
+/-- The submonoid generated by an element is a group if that element has finite order. -/
+@[to_additive "The additive submonoid generated by an element is an additive group
+if that element has finite order."]
+noncomputable abbrev IsOfFinOrder.groupPowers (hx : IsOfFinOrder x) : Group (powers x) where
```

Comment:
As discussed on Zulip, I think we should split this into
```suggestion
abbrev Submonoid.groupPowersOfPow (hn : 0 < n) (hx : x ^ n = 1) : Group (powers x) where
```
and
```suggestion
noncomputable abbrev IsOfFinOrder.groupPowers (hx : IsOfFinOrder x) : Group (powers x) :=
   Submonoid.groupPowersOfPow
     ((isOfFinOrder_iff_pow_eq_one x).1 hx).choose_spec.1
     ((isOfFinOrder_iff_pow_eq_one x).1 hx).choose_spec.2
```

Reply:
Done (without seeing this comment prior).